include/soci/soci-backend.h

```diff
@@ -319,6 +322,68 @@ class blob_backend
     SOCI_NOT_COPYABLE(blob_backend)
 };
 
+/**
+ * This Blob implementation uses an explicit buffer that is read from and written to, instead of
+ * directly communicating with the underlaying database.
+ * Thus, it is intended to be used whenever the underlying database does not offer a more efficient
+ * way of dealing with BLOBs.
+ */
+class trivial_blob_backend : public blob_backend
+{
+public:
+    std::size_t get_len() override { return buffer_.size(); }
+
+    std::size_t read_from_start(char* buf, std::size_t toRead,
+        std::size_t offset = 0) override
+    {
+        if (offset > buffer_.size() || (offset ==  buffer_.size() && offset > 0))
+        {
+            throw soci_error("Can't read past-the-end of BLOB data.");
+        }
+
+        toRead = (std::min)(toRead, buffer_.size() - offset);
```

Comment:
Because the Windows API is ridiculous and defines function-like macros named `min` and `max` that clash with the function names of the std functions: https://stackoverflow.com/q/5004858

Reply:
I don't know where is `windows.h` getting included from, but it's traditional to predefine `NOMINMAX` before doing it, which is a better idea than to uglify all the code.

Reply:
This is a public header file so we don't have control over whether or not these Windows macros have been defined in the translation unit prior to including this header.
I think this is actually exactly what happens somewhere in the Windows test suite of SOCI, which is how I discovered the issue.

Reply:
It would be preferable to add `#undef min` to doing this but I've just realized that this class should _not_ be in a public header in the first place: it's only used inside the library and will never be used by the application, so why do we define it here? It should be moved to a private header, IMO.

Reply:
> It would be preferable to add #undef min to doing

no - code after having included this header could be relying on `min` being defined.

> I've just realized that this class should not be in a public header in the first place: it's only used inside the library and will never be used by the application, so why do we define it here? It should be moved to a private header, IMO.

I added it here to make it available to all backends that need it. It would indeed be better to define it in a private header though. I'll move it accordingly.